app/src/main/java/com/javahelp/frontend/MainActivity.java

```diff
@@ -2,59 +2,86 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProvider;
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
 import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
 
+import com.javahelp.BR;
 import com.javahelp.R;
+import com.javahelp.databinding.ActivityMainBinding;
 
-public class MainActivity extends AppCompatActivity {
 
-    private Button buttonLogin;
-    private  Button buttoncr;
-    private  Button buttonpr;
+public class MainActivity extends AppCompatActivity {
+    MyViewModel myViewModel;
+    ActivityMainBinding binding;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        //setContentView(R.layout.activity_main);
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
+        myViewModel = new ViewModelProvider(this).get(MyViewModel.class);
+        binding.setData(myViewModel);
+        binding.setLifecycleOwner(this);
 
-        //find the login button by its id
-        buttonLogin = (Button) findViewById(R.id.bot_login);
-        //find the client register button by its id
-        buttoncr = (Button) findViewById(R.id.bot_c_reg);
-        //find the provider register button by its id
-        buttonpr = (Button) findViewById(R.id.bot_p_reg);
 
-        // link the button to the second activity(front page)
-        buttonLogin.setOnClickListener(new View.OnClickListener() {
+        // link the button to the second acticity(front page)
+        binding.botLogin.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                Intent intent = new Intent(MainActivity.this,FrontPageActivity.class);
-                startActivity(intent);
+                Intent intent = null;
+                if (usernameChecker()) {
+                    intent = new Intent(MainActivity.this,FrontPageActivity.class);
+                    startActivity(intent);
+                    finish();
+                }
+                else{
+                    String inv ="username invalid";
+                    Toast.makeText(getApplicationContext(), inv , Toast.LENGTH_SHORT).show();
+                }
             }
         });
 
-        // link the button to the second activity(front page)
-        buttonpr.setOnClickListener(new View.OnClickListener() {
+        // link the button to the provider Registeration page
+        binding.botPReg.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                Intent intent = new Intent(MainActivity.this,ProviderRegistrationActivity.class);
+                Intent intent = null;
```

Comment:
I'm not really sure why these two cannot be shortened into one.

Reply:
Same, I think, in that case, it would exceed the character limit in a line

Reply:
You could space it out if you like `Intent intent = new Intent(MainActivity.this, 
                                                        ProviderRegistrationActivity.class);`

Reply:
but it is not a big deal, so don't worry